src/ASM/LR/LRSplineField.C

```diff
@@ -36,21 +38,21 @@ bool LRSplineField::evalMapping (const LR::LRSplineSurface& surf,
   Matrix3D d2Ndu2;
   if (Hess) {
     Go::BasisDerivsSf2 spline2;
-    surf.computeBasis(x.u,x.v,spline2,iel);
+    if (is_rational)
+      ASMu2D::computeBasisNurbs(x.u,x.v,spline2,iel,surf);
+    else
+      surf.computeBasis(x.u,x.v,spline2,iel);
     SplineUtils::extractBasis(spline2, N, dNdu, d2Ndu2);
   } else {
     Go::BasisDerivsSf spline;
-    surf.computeBasis(x.u,x.v,spline,iel);
+    if (is_rational)
+      ASMu2D::computeBasisNurbs(x.u,x.v,spline,iel,surf);
+    else
+      surf.computeBasis(x.u,x.v,spline,iel);
     SplineUtils::extractBasis(spline, N, dNdu);
   }
 
-  Xnod.resize(2,elm->nBasisFunctions());
-  size_t i = 1;
-  for (const LR::Basisfunction* f : elm->support()) {
-    for (size_t j = 1; j <= 2; ++j)
-      Xnod(j,i) = f->cp(j-1);
-    ++i;
-  }
+  ASMu2D::getCoordinates(Xnod,surf.dimension() - (is_rational ? 1 : 0),surf,iel+1);
```

Comment:
since a bool true=1 and false=0 one could just skip the ? here and just do `surf.dimension()-is_rational`, or will clang complain about mismatching types in operator- in that case.

Reply:
seems fine.